crates/optimism/payload/src/builder.rs

```diff
@@ -377,14 +378,22 @@ impl<Txs> OpBuilder<'_, Txs> {
         state.merge_transitions(BundleRetention::Reverts);
 
         let withdrawals_root = if ctx.is_isthmus_active() {
+            let hashed_storage_updates =
+                state.bundle_state.state().get(&ADDRESS_L2_TO_L1_MESSAGE_PASSER).map(|account| {
+                    // block contained withdrawals transactions, use predeploy storage updates from
+                    // execution
+                    HashedStorage::from_plain_storage(
+                        account.status,
+                        account.storage.iter().map(|(slot, value)| (slot, &value.present_value)),
+                    )
+                });
+
             // withdrawals root field in block header is used for storage root of L2 predeploy
             // `l2tol1-message-passer`
-            Some(
-                state
-                    .database
-                    .as_ref()
-                    .storage_root(ADDRESS_L2_TO_L1_MESSAGE_PASSER, Default::default())?,
-            )
+            Some(state.database.as_ref().storage_root(
+                ADDRESS_L2_TO_L1_MESSAGE_PASSER,
+                hashed_storage_updates.unwrap_or_default(),
+            )?)
```

Comment:
this most likely does redundant IO because we might traverse this path during state root computation. please, leave a comment saying this. the proper way would be to include `state_root` in the `MultiProof` struct and call `state_provider.multiproof` with the target instead of `state_provider.root` when sealing a block

Reply:
opened an issue https://github.com/paradigmxyz/reth/issues/14492

Reply:
thank you @rkrasiuk !